plugins/module_utils/core.py

```diff
@@ -255,22 +265,24 @@ def present(
     ) -> bool:
         results = {"changed": False, "result": {}}
         create_only_params = create_only_params or []
+        resource = None
         identifier: Dict = {}
 
-        resource = None
         if self.module.params.get("identifier"):
             identifier = self.module.params.get("identifier")
         else:
-            identifier = self.get_identifier(identifier, primary_identifier)
+            identifier = self.get_identifier(primary_identifier)
+
         try:
             resource = self.client.get_resource(
                 TypeName=type_name, Identifier=identifier, aws_retry=True
             )
             results = self.update_resource(resource, params, create_only_params)
         except self.client.exceptions.ResourceNotFoundException:
             if self.module.params.get("identifier"):
+                _primary_identifier = [id.split("/")[-1] for id in primary_identifier]
                 self.module.fail_json(
-                    f"""You must specify both {*primary_identifier, } to create a new resource.
+                    f"""You must specify together {*_primary_identifier, } to create a new resource.
```

Comment:
I'm not sure I understand this change. Is it because there may be more than two?

Reply:
There might be, that's why I wanted to make it more generic.

Reply:
OK, I would suggest a change to wording in that case. Something like `You must specify all of the following to create a new resource: {*_primary_identifier, }.`

Reply:
Done, thanks!